### PR TITLE
Sharp Memory Display displayio support (v2)

### PR DESCRIPTION
**Pull request by jepler**
Typical construction:
```
bus = board.SPI()
framebuffer = sharpdisplay.SharpMemoryFramebuffer(bus, board.D2, 400, 240)
display = framebufferio.FramebufferDisplay(framebuffer, auto_refresh = True)
```

Full demo, which happens to be set up for Particle Xenon, CS on D2: [code.txt](https://github.com/adafruit/circuitpython/files/5059274/code.txt)

### CODE REVIEW

**Comment by jepler** [comment]
I should also retest with rgbmatrix, as dirty row tracking could affect it in some unantipated way.  Testing so far is only with the 400x240 sharp display on nrf52840 though

File: ports/nrf/Makefile [hunk 1]
```diff
@@ -100,8 +100,8 @@ CFLAGS += $(OPTIMIZATION_FLAGS)
 CFLAGS += $(INC) -Wall -Werror -std=gnu11 -nostdlib -fshort-enums $(BASE_CFLAGS) $(CFLAGS_MOD) $(COPT)
 
 # Undo some warnings.
-# nrfx uses undefined preprocessor variables quite casually, so we can't do warning checks for these.
-CFLAGS += -Wno-undef
+## nrfx uses undefined preprocessor variables quite casually, so we can't do warning checks for these.
+#CFLAGS += -Wno-undef
```
Comment (tannewt) [inline ports/nrf/Makefile:104]: Want to move this comment below to state it only applies to nrfx?

File: shared-module/framebufferio/FramebufferDisplay.h [hunk 2]
```diff
@@ -48,44 +48,64 @@ typedef struct {
     uint64_t last_refresh_call;
     uint16_t native_frames_per_second;
     uint16_t native_ms_per_frame;
+    uint16_t first_pixel_offset, row_stride;
```
Comment (tannewt) [inline shared-module/framebufferio/FramebufferDisplay.h:51]:
```suggestion
    uint16_t first_pixel_offset;
    uint16_t row_stride;
```

File: shared-module/sharpdisplay/SharpMemoryFramebuffer.c [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+    // set chip select high
+    common_hal_digitalio_digitalinout_set_value(&self->chip_select, true);
+
+    // output the toggling signal
+    uint8_t *data = self->bufinfo.buf;
+    data[0] ^= SHARPMEM_BIT_VCOM_LSB;
+
+#define PARTIAL_UPDATES (1)
+#if PARTIAL_UPDATES
```
Comment (tannewt) [inline shared-module/sharpdisplay/SharpMemoryFramebuffer.c:187]: Want to just turn this on for good?

File: ports/nrf/Makefile (hunk 1 above)
Comment (jepler) [reply ports/nrf/Makefile:104]: will do

File: shared-module/sharpdisplay/SharpMemoryFramebuffer.c (hunk 3 above)
Comment (jepler) [reply shared-module/sharpdisplay/SharpMemoryFramebuffer.c:187]: will do

**Pull request merged**